crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -149,26 +130,35 @@ pub fn impl_reflect_value(input: TokenStream) -> TokenStream {
 #[proc_macro]
 pub fn impl_reflect_struct(input: TokenStream) -> TokenStream {
     let ast = parse_macro_input!(input as DeriveInput);
-    let derive_data = match ReflectDeriveData::from_input(&ast) {
+    let derive_data = match ReflectDerive::from_input(&ast) {
         Ok(data) => data,
         Err(err) => return err.into_compile_error().into(),
     };
 
-    let impl_struct: proc_macro2::TokenStream = impls::impl_struct(&derive_data).into();
-    let impl_from_struct: proc_macro2::TokenStream = from_reflect::impl_struct(&derive_data).into();
-
-    TokenStream::from(quote! {
-        #impl_struct
-
-        #impl_from_struct
-    })
+    match derive_data {
+        ReflectDerive::Struct(struct_data) => {
+            let impl_struct: proc_macro2::TokenStream = impls::impl_struct(&struct_data).into();
+            let impl_from_struct: proc_macro2::TokenStream =
+                from_reflect::impl_struct(&struct_data).into();
+
+            TokenStream::from(quote! {
+                #impl_struct
+
+                #impl_from_struct
+            })
+        }
+        _ => syn::Error::new(
+            ast.span(),
+            "impl_reflect_struct is only supported for standard structs",
```

Comment:
This error message is not very helpful. What's a "standard" struct here, and when would this occur?

Reply:
Are the updated messages better?

Reply:
Much better, thank you.